reporter/base_reporter.go

```diff
@@ -78,7 +78,7 @@ func (b *baseReporter) FrameKnown(frameID libpf.FrameID) bool {
 		pdata.FramesCacheLifetime); exists {
 		frameMap := frameMapLock.RLock()
 		defer frameMapLock.RUnlock(&frameMap)
-		_, known = (*frameMap)[frameID.AddressOrLine()]
+		_, known = (*frameMap).GetAndRefresh(frameID.AddressOrLine(), pdata.FrameMapLifetime)
```

Comment:
@rockdaboot Not for this PR but it seems clumsy to have to specify the lifetime every time `GetAndRefresh` is called if `SetLifetime` has been previously called to set one (if the lifetime we're specifying isn't different from the one we've previously set). Assuming that this is the common case, maybe we should optimize for it:

`GetAndRefresh` can use the `LRU` lifetime and a new method (e.g. `GetAndRefreshWithLifetime`) could be implement to support custom per-value lifetimes.